pytest/pytest_shim.py

```diff
@@ -0,0 +1,71 @@
+"""A shim for executing pytest that supports test filtering, sharding, and more.
+
+Copied from https://github.com/caseyduquettesc/rules_python_pytest/blob/331e0e511130cf4859b7589a479db6c553974abf/python_pytest/pytest_shim.py
+"""
+
+import sys
+import os
+
+import pytest
+
+
+if __name__ == "__main__":
+    pytest_args = ["--ignore=external"]
+
+    args = sys.argv[1:]
+    # pytest < 8.0 runs tests twice if __init__.py is passed explicitly as an argument.
+    # Remove any __init__.py file to avoid that.
+    # pytest.version_tuple is available since pytest 7.0
+    # https://github.com/pytest-dev/pytest/issues/9313
+    if not hasattr(pytest, "version_tuple") or pytest.version_tuple < (8, 0):
+        args = [arg for arg in args if arg.startswith("-") or os.path.basename(arg) != "__init__.py"]
+
+    if os.environ.get("XML_OUTPUT_FILE"):
+        pytest_args.append("--junitxml={xml_output_file}".format(xml_output_file=os.environ.get("XML_OUTPUT_FILE")))
+
+    # Handle test sharding - requires pytest-shard plugin.
```

Comment:
Can the presence of the plugin be detected, and an error printed if it's missing?

Reply:
Can I file and leave as a follow-up? Right now I'm looking for parity with v1.1.1 https://registry.bazel.build/modules/caseyduquettesc_rules_python_pytest